name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TradeRepublicPDFExtractor.java

```diff
@@ -1037,7 +1040,124 @@ private void addDeliveryInOutBoundTransaction()
+    {
+        DocumentType type = new DocumentType("(STEUERLICHER UMTAUSCH)", (context, lines) -> {
+            Pattern pDate = Pattern.compile("^(.*) DATUM (?<date>[\\d]{2}\\.[\\d]{2}\\.[\\d]{4})$");
+            context.put("transactionPosition", "0");
+
+            for (String line : lines)
+            {
+                Matcher mDate = pDate.matcher(line);
+                if (mDate.matches())
+                    context.put("date", mDate.group("date"));
+
+            }
+        });
+        this.addDocumentTyp(type);
+
+        Transaction<PortfolioTransaction> pdfTransaction = new Transaction<>();
+
+        Block firstRelevantLine = new Block("(^|^[\\d] )(Einbuchung"
+                        + "|Ausbuchung) .* "
+                        + "([\\.,\\d]+ Stk\\.)$");
+        type.addBlock(firstRelevantLine);
+        firstRelevantLine.set(pdfTransaction);
+
+        pdfTransaction //
+
+                .subject(() -> {
+                    PortfolioTransaction portfolioTransaction = new PortfolioTransaction();
+                    portfolioTransaction.setType(PortfolioTransaction.Type.DELIVERY_OUTBOUND);
+                    return portfolioTransaction;
+                })
+
+                // Is type --> "Einbuchung" change from DELIVERY_OUTBOUND to DELIVERY_INBOUND
+                .section("type")
+                .match("(^|^[\\d] )(?<type>"
+                                + "(Einbuchung|Ausbuchung)"
+                                + ") .* ([\\.,\\d]+ Stk\\.)$")
+                .assign((t, v) -> {
+                    if ("Einbuchung".equals(v.get("type")))
+                    {
+                        t.setType(PortfolioTransaction.Type.DELIVERY_INBOUND);
+                        t.setNote("Einstandskurs fehlt");
+                    }
+                })
 
+                // @formatter:off
+                // 1 Ausbuchung AMUN.GOV.BD EO BR.IG 1-3 U.ETF 50 Stk.
+                // Actions au Porteur o.N.
+                // FR0010754135
+                //
+                // 1 Einbuchung MUL-LYX.EO Gov.Bd 1-3Y(DR)U.E. 67,541 Stk.
+                // Nam.-An. Acc o.N.
+                // LU1650487413
+                // @formatter:on
+                .section("position", "name", "shares", "nameContinued", "isin")
+                .match("^(?<position>[\\d]) (Einbuchung|Ausbuchung) (?<name>.*) (?<shares>[\\.,\\d]+) Stk\\.$")
+                .match("^(?<nameContinued>.*)$")
+                .match("^(?<isin>[A-Z]{2}[A-Z0-9]{9}[0-9])$")
+                .assign((t, v) -> {
+                    Map<String, String> context = type.getCurrentContext();
+                    context.put("position", v.get("position"));
+                    t.setDateTime(asDate(context.get("date")));
+                    t.setShares(asShares(v.get("shares")));
+                    t.setSecurity(getOrCreateSecurity(v));
+                    t.setCurrencyCode(asCurrencyCode(t.getSecurity().getCurrencyCode()));
+                    t.setAmount(0L);
```

Comment:
Is this variable "position" reused?

With the change for reading the date, `.documentContext("date")` is now used. access to the variable `v.get("date")` is now possible. In addition, we now check whether this exists. 

```suggestion
                .section("name", "shares", "nameContinued", "isin")
                .documentContext("date") //
                .match("^[\\d] (Einbuchung|Ausbuchung) (?<name>.*) (?<shares>[\\.,\\d]+) Stk\\.$")
                .match("^(?<nameContinued>.*)$")
                .match("^(?<isin>[A-Z]{2}[A-Z0-9]{9}[0-9])$")
                .assign((t, v) -> {
                    t.setDateTime(v.get("date"));
                    t.setShares(asShares(v.get("shares")));
                    t.setSecurity(getOrCreateSecurity(v));
                    t.setCurrencyCode(asCurrencyCode(t.getSecurity().getCurrencyCode()));
                    t.setAmount(0L);
```